From e12a6b22bfc74fdbd58a516c0df0723a509f16c2 Mon Sep 17 00:00:00 2001
From: "Friedrich W. H. Kossebau" <kossebau@kde.org>
Date: Fri, 6 Jul 2018 18:11:25 +0200
Subject: [PATCH] Update D-Bus xml files to use "Out*" for signal type Qt
 annotations

Since Qt 5.6.3/5.7.1 qdbusxml2cpp favours as name for signals argument
annotations the name "org.qtproject.QtDBus.QtTypeName.Out*", matching
the "out" direction nature of the arguments.
".In*" is still supported for backward compat, but with a warning.

Upstream-Status: Backport [https://github.com/KDE/bluez-qt/commit/d52c2284f90dec6534e88727d0a0d94abe1f5f4d]]
Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 .../autotests/interfaces/org.freedesktop.DBus.Properties.xml    | 2 +-
 bluez-qt/src/interfaces/org.freedesktop.DBus.ObjectManager.xml  | 2 +-
 bluez-qt/src/interfaces/org.freedesktop.DBus.Properties.xml     | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/bluez-qt/autotests/interfaces/org.freedesktop.DBus.Properties.xml b/bluez-qt/autotests/interfaces/org.freedesktop.DBus.Properties.xml
index 20ddc11..688ed10 100644
--- a/bluez-qt/autotests/interfaces/org.freedesktop.DBus.Properties.xml
+++ b/bluez-qt/autotests/interfaces/org.freedesktop.DBus.Properties.xml
@@ -22,7 +22,7 @@
       <arg type="s" name="interface_name"/>
       <arg type="a{sv}" name="changed_properties"/>
       <arg type="as" name="invalidated_properties"/>
-      <annotation name="org.qtproject.QtDBus.QtTypeName.In1" value="QVariantMap" />
+      <annotation name="org.qtproject.QtDBus.QtTypeName.Out1" value="QVariantMap" />
     </signal>
   </interface>
 </node>
diff --git a/bluez-qt/src/interfaces/org.freedesktop.DBus.ObjectManager.xml b/bluez-qt/src/interfaces/org.freedesktop.DBus.ObjectManager.xml
index 5a962c3..45770b8 100644
--- a/bluez-qt/src/interfaces/org.freedesktop.DBus.ObjectManager.xml
+++ b/bluez-qt/src/interfaces/org.freedesktop.DBus.ObjectManager.xml
@@ -9,7 +9,7 @@
     <signal name="InterfacesAdded">
       <arg name="object" type="o"/>
       <arg name="interfaces" type="a{sa{sv}}"/>
-      <annotation name="org.qtproject.QtDBus.QtTypeName.In1" value="QVariantMapMap"/>
+      <annotation name="org.qtproject.QtDBus.QtTypeName.Out1" value="QVariantMapMap"/>
     </signal>
     <signal name="InterfacesRemoved">
       <arg name="object" type="o"/>
diff --git a/bluez-qt/src/interfaces/org.freedesktop.DBus.Properties.xml b/bluez-qt/src/interfaces/org.freedesktop.DBus.Properties.xml
index 20ddc11..688ed10 100644
--- a/bluez-qt/src/interfaces/org.freedesktop.DBus.Properties.xml
+++ b/bluez-qt/src/interfaces/org.freedesktop.DBus.Properties.xml
@@ -22,7 +22,7 @@
       <arg type="s" name="interface_name"/>
       <arg type="a{sv}" name="changed_properties"/>
       <arg type="as" name="invalidated_properties"/>
-      <annotation name="org.qtproject.QtDBus.QtTypeName.In1" value="QVariantMap" />
+      <annotation name="org.qtproject.QtDBus.QtTypeName.Out1" value="QVariantMap" />
     </signal>
   </interface>
 </node>
